src/negotiation/maker_negotiator.ts

```diff
@@ -29,44 +35,58 @@ export class MakerNegotiator {
     return this.ordersById[orderId];
   }
 
-  public acceptOrder(
-    // @ts-ignore
-    order: Order
-  ): ExecutionParams | undefined {
+  public getExecutionParams(): ExecutionParams {
+    return this.executionParams;
+  }
+
+  public acceptOrder(swapId: string, order: Order) {
     // Fire the auto-accept of the order in the background
     (async () => {
       try {
-        await this.tryAcceptSwap(order, this.tryParams);
+        await this.tryAcceptSwap(swapId, order, this.tryParams);
       } catch (error) {
         console.log("Could not accept the swap");
       }
     })();
-    return this.executionParams;
-  }
-
-  public addOrder(order: Order) {
-    this.ordersByTradingPair[order.tradingPair] = order;
-    this.ordersById[order.id] = order;
   }
+  // End of methods related to the negotiation protocol
 
   private async tryAcceptSwap(
+    swapId: string,
     order: Order,
-    { timeout, tryInterval }: TryParams
+    { maxTimeoutSecs, tryIntervalSecs }: TryParams
   ) {
-    return timeoutPromise(timeout, this.acceptSwap(order, tryInterval));
+    return timeoutPromise(
+      maxTimeoutSecs * 1000,
+      this.acceptSwap(swapId, order, tryIntervalSecs)
+    );
   }
 
-  private async acceptSwap(order: Order, tryInterval: number) {
+  private async acceptSwap(
+    swapId: string,
+    order: Order,
+    tryIntervalSecs: number
+  ) {
     while (true) {
-      await sleep(tryInterval);
+      await sleep(tryIntervalSecs * 1000);
 
-      const swap = await this.comitClient.retrieveSwapByOrder(order);
+      const swap = await this.comitClient.retrieveSwapById(swapId);
```

Comment:
A more general statement, because I just stumbled about it: 
I feel the negotiation should be better encapsulated in the SDK. I feel the `MakerNegotiator` should not know about the `ComitClient`, but the swap should be fetched and then passed from the outside. If the `MakerNegotiator` knows about the `ComitClient` that couples the negotiation and execution tightly. If we later plan to take the execution into a separate module, would the common denominator be the `ComitClient`?

You don't have to give me a lengthy reply @D4nte - we can discuss this tomorrow, just wanted to write down my thoughts in the right place.

Reply:
This is designed as per @bonomat's original request to automate the swap request and acceptance.

Reply:
I will stop putting such comments in, guess we have to roll with something and think about the bigger picture after we have feedback :)